packages/nuxt/src/imports/module.ts

```diff
@@ -228,19 +234,66 @@ function addDeclarationTemplates (ctx: Unimport, options: Partial<ImportsOptions
 
   addTypeTemplate({
     filename: 'imports.d.ts',
-    getContents: async ({ nuxt }) => toExports(await ctx.getImports(), nuxt.options.buildDir, true),
+    getContents: async ({ nuxt }) => {
+      const allImports = await ctx.getImports()
+      // Exclude shared imports to avoid duplication
+      const nonSharedImports = allImports.filter(imp => !isSharedImport(imp))
+      return [
+        '/// <reference path="./types/shared.d.ts" />',
+        toExports(nonSharedImports, nuxt.options.buildDir, true),
+      ].join('\n')
+    },
   })
 
   addTypeTemplate({
     filename: 'types/imports.d.ts',
     getContents: async () => {
-      const imports = await ctx.getImports()
-      await cacheImportPaths(imports)
-      return '// Generated by auto imports\n' + (
+      const allImports = await ctx.getImports()
+      await cacheImportPaths(allImports)
+
+      // Exclude shared imports to avoid duplication
+      const nonSharedImports = allImports.filter(imp => !isSharedImport(imp))
+
+      // Create a temporary context with only non-shared imports
+      const nonSharedCtx = createUnimport({
```

Comment:
this is likely to be non performant, as it creates a new unimport context every time the file is generated